### PR TITLE
#136: Added page count generation to pdf files

### PR DESCRIPTION
**Pull request by CloudLevi**
Implemented #136

### CODE REVIEW

File: app/src/main/java/space/taran/arknavigator/mvp/model/repo/extra/DocumentMetaExtra.kt [hunk 1]
```diff
@@ -1,11 +1,34 @@
...
+                ParcelFileDescriptor.MODE_READ_ONLY
+            )
+            val pdfRenderer = PdfRenderer(parcelFileDescriptor)
+            val totalPages = pdfRenderer.pageCount
+            if (totalPages > 0) {
+                result[MetaExtraTag.PAGES] = totalPages.toLong()
+                return ResourceMetaExtra(result)
+            }
+        } catch (e: FileNotFoundException) {
+            e.printStackTrace()
```
Comment (kirillt) [inline app/src/main/java/space/taran/arknavigator/mvp/model/repo/extra/DocumentMetaExtra.kt:30]: Would be good to add `Log.e` with information about what we were doing when exception happened.

File: app/src/main/java/space/taran/arknavigator/mvp/model/repo/extra/DocumentMetaExtra.kt [hunk 2]
```diff
@@ -1,11 +1,34 @@
...
-    fun extract(path: Path): ResourceMetaExtra? = null
+    fun extract(path: Path): ResourceMetaExtra? {
+        val result = mutableMapOf<MetaExtraTag, Long>()
+        if (extension(path) != "pdf") return null
+        try {
+            val parcelFileDescriptor = ParcelFileDescriptor.open(
+                path.toFile(),
+                ParcelFileDescriptor.MODE_READ_ONLY
+            )
+            val pdfRenderer = PdfRenderer(parcelFileDescriptor)
```
Comment (kirillt) [inline app/src/main/java/space/taran/arknavigator/mvp/model/repo/extra/DocumentMetaExtra.kt:23]: If this and following lines can not throw `FileNotFoundException`, then I would close `try-catch` before it.

**Comment by vedranqa** [comment]
@kirillt  Seems fine
https://prnt.sc/26r86fb
https://prnt.sc/26r88cp

**Review by kirillt — APPROVED** [review]
Good

**Pull request merged**